apps/dashboard/pkg/migration/schemaversion/v34.go

```diff
@@ -0,0 +1,294 @@
+			if statString, ok := stat.(string); ok {
+				// Create a copy of the original query
+				newQuery := make(map[string]interface{})
+				for k, v := range t {
+					if k != "statistics" {
+						newQuery[k] = v
+					}
+				}
+
+				// Set the single statistic
+				newQuery["statistic"] = statString
+
+				if i == 0 {
+					// First query replaces the original
+					newTargets = append(newTargets, newQuery)
+				} else {
+					// Additional queries get new refIds and are added at the end
+					if refId, ok := newQuery["refId"].(string); ok {
+						newQuery["refId"] = generateNextRefId(append(targets, additionalTargets...), refId, len(additionalTargets))
+					}
+					additionalTargets = append(additionalTargets, newQuery)
+				}
+			}
+		}
+	}
+
+	// Append additional queries at the end
+	panel["targets"] = append(newTargets, additionalTargets...)
+}
+
+// migrateCloudWatchAnnotationQueries migrates CloudWatch annotation queries that use multiple statistics.
+func migrateCloudWatchAnnotationQueries(dashboard map[string]interface{}) {
+	annotations, ok := dashboard["annotations"].(map[string]interface{})
+	if !ok {
+		return
+	}
+
+	annotationsList, ok := annotations["list"].([]interface{})
+	if !ok {
+		return
+	}
+
+	var additionalAnnotations []interface{}
+
+	for i, annotation := range annotationsList {
+		a, ok := annotation.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		if !isLegacyCloudWatchAnnotationQuery(a) {
+			continue
+		}
+
+		// Check if this annotation has multiple statistics
+		statistics, hasStatistics := a["statistics"].([]interface{})
+		if !hasStatistics || len(statistics) <= 1 {
+			// Convert single statistic to proper format
+			if hasStatistics && len(statistics) == 1 {
+				if stat, ok := statistics[0].(string); ok {
+					// Create new annotation with single statistic
+					newAnnotation := make(map[string]interface{})
+					for k, v := range a {
+						if k != "statistics" {
+							newAnnotation[k] = v
+						}
+					}
+					newAnnotation["statistic"] = stat
+					annotationsList[i] = newAnnotation
+				}
+			}
+			continue
+		}
+
+		// Split annotation with multiple statistics into separate annotations
+		// Create new annotations for each statistic, replace original with first one
+		originalName, hasName := a["name"].(string)
+
+		for j, stat := range statistics {
+			if statString, ok := stat.(string); ok {
+				// Create new annotation for this statistic
+				newAnnotation := make(map[string]interface{})
+				for k, v := range a {
+					if k != "statistics" {
+						newAnnotation[k] = v
+					}
+				}
+
+				// Set the single statistic
+				newAnnotation["statistic"] = statString
+
+				// Set the name with statistic suffix if multiple statistics
+				if len(statistics) > 1 && hasName {
+					newAnnotation["name"] = originalName + " - " + statString
+				}
+
+				if j == 0 {
+					// Replace the original annotation with the first new one
+					annotationsList[i] = newAnnotation
```

Comment:
Also here -- if the string conversion fails, will this cause an issue when the original annotation is not replaced?